SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForInterestActor.cpp

```diff
@@ -128,16 +129,17 @@ void ASpatialTestInitialOnlyForInterestActor::PrepareTest()
 		FinishStep();
 	});
 
-	AddStep(TEXT("Check changed value."), FWorkerDefinition::Client(1), nullptr, [this]() {
+	AddStep(TEXT("Check changed value."), FWorkerDefinition::Client(1), nullptr, nullptr, [this](float DeltaTime) {
 		TArray<AActor*> SpawnActors;
 		UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+		AssertEqual_Int(SpawnActors.Num(), 1, TEXT("There should be exactly one InitialOnly actor in the world."));
 		for (AActor* Actor : SpawnActors)
 		{
 			ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
-			if (SpawnActor != nullptr)
+			if (AssertIsValid(SpawnActor, TEXT("SpawnActor should be valid.")))
 			{
-				AssertTrue(SpawnActor->Int_Initial == 1, TEXT("Check Actor.Int_Initial value."));
-				AssertTrue(SpawnActor->Int_Replicate == 2, TEXT("Check Actor.Int_Replicate value."));
+				RequireEqual_Int(SpawnActor->Int_Initial, 1, TEXT("Check Actor.Int_Initial value."));
```

Comment:
I'm not sure this should be a require - it should always be 1. If the require passes on the first tick (because it just hasn't replicated), will it actually fail on a subsequent tick if the initial only value does get updated?
Perhaps switch to an assert check that fires on when the INT_replicate value has been updated to 2.

Reply:
For the first point, it needs to be a require, because the previous step is setting the values on the server, but this step is checking them on the client. There's no guarantee that the property replication will arrive on the client faster than the RPC which signals the client to start the next step in the test. Usually, we deal with this using the requires to wait until the condition is satisfied.
For your second point, both requires have to be satisfied at the same time for the test to successfully finish the step. So it's kind of an implementation detail, but we will basically wait until Int_Replicate becomes 2, and Int_Initial has to be 1 at the same time. The alternative would be the "wait-so-bad-thing-does-not-happen" pattern, where we could introduce a wait here of e.g. 1 second to "make sure" that the replication for the Int_Initial never comes.

Reply:
Cool - because the requires are reset each time they're called this is fine 👍